controllers/constraintstatus_controller.go

```diff
@@ -0,0 +1,371 @@
+	config := &v1alpha1.Config{}
+	err = r.Get(ctx, types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      "config",
+	}, config)
+
+	if apierrors.IsNotFound(err) {
+		config = &v1alpha1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "config",
+				Namespace: r.Namespace,
+			},
+		}
+
+		err = controllerutil.SetOwnerReference(gatekeeper, config, r.Scheme)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		err = r.Create(ctx, config)
+		if err != nil {
+			log.Error(err, "Fail to Create Config resource, requeue")
+
+			return reconcile.Result{}, err
+		}
+
+		log.Info("The Gatekeeper's Config resource is just created")
+	}
+
+	if err != nil && !apierrors.IsNotFound(err) {
+		return reconcile.Result{}, err
+	}
+
+	constraintPodStatus := &v1beta1.ConstraintPodStatus{}
+
+	err = r.Get(ctx, request.NamespacedName, constraintPodStatus)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(1).Info("Cannot find the ConstraintPodStatus")
+
+			err = r.handleDeleteEvent(ctx, request.Name, config)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			return reconcile.Result{}, nil
+		}
+		// Requeue
+		return reconcile.Result{}, err
+	}
+
+	labels := constraintPodStatus.GetLabels()
+
+	constraintKind := labels["internal.gatekeeper.sh/constraint-kind"]
+	constraintName := labels["internal.gatekeeper.sh/constraint-name"]
+
+	constraintGVR := schema.GroupVersionResource{
+		Group:    "constraints.gatekeeper.sh",
+		Version:  "v1beta1",
+		Resource: strings.ToLower(constraintKind),
+	}
+
+	constraint, err := r.DynamicClient.Resource(constraintGVR).Get(ctx, constraintName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			r.Log.Info("The Constraint was not found", "constraintName:", constraintName)
+
+			return reconcile.Result{}, nil
+		}
+
+		return reconcile.Result{}, err
+	}
+
+	constraintMatchKinds, _, err := unstructured.NestedSlice(constraint.Object, "spec", "match", "kinds")
+	if err != nil {
+		r.Log.V(1).Info("There is no provided kinds in contsraint", "constraintName:", constraintName)
+
+		return reconcile.Result{}, nil //nolint:nilerr
+	}
+
+	constraintSyncOnlyEntries, err := r.getSyncOnlys(constraintMatchKinds)
+	if err != nil {
+		if errors.Is(err, ErrNotFoundDiscovery) {
+			r.Log.V(1).Info("Cannot find matched discovery. Requeue after 3 mins")
+
+			requeueTime = time.Second * 10
+		} else {
+			log.Error(err, "Error to get matching kind and apigroup")
+
+			return reconcile.Result{}, err
+		}
+	}
+
+	r.ConstraintToSyncOnly[request.Name] = *constraintSyncOnlyEntries
+
+	updatedSyncOnly := r.getUniqSyncOnly()
+
+	if reflect.DeepEqual(updatedSyncOnly, config.Spec.Sync.SyncOnly) {
+		r.Log.V(1).Info("There is no changes detected. Cancel Update")
+
+		return reconcile.Result{RequeueAfter: requeueTime}, nil
+	}
+
+	config.Spec.Sync.SyncOnly = updatedSyncOnly
+
+	err = r.Update(ctx, config, &client.UpdateOptions{})
+	if err != nil {
+		return reconcile.Result{}, err
```

Comment:
You can still update but have encountered a discovery error.
```suggestion
		return reconcile.Result{RequeueAfter: requeueTime}, err
```

Reply:
```
return reconcile.Result{}, err
```